web/src/compose_validate.ts

```diff
@@ -889,9 +884,18 @@ export function validate_message_length($container: JQuery): boolean {
     const $textarea = $container.find<HTMLTextAreaElement>(".message-textarea");
     // Match the behavior of compose_state.message_content of trimming trailing whitespace
     const text = $textarea.val()!.trimEnd();
-    if (text.length > realm.max_message_length) {
+    const message_too_long_for_compose = text.length > realm.max_message_length;
+    set_message_too_long_for_compose(message_too_long_for_compose);
+    if (message_too_long_for_compose) {
         $textarea.addClass("flash");
         setTimeout(() => $textarea.removeClass("flash"), 1500);
+        const $banner_container = $("#compose_banners");
+        compose_banner.show_error_message(
+            get_message_too_long_for_compose_error(),
+            compose_banner.CLASSNAMES.exceeded_message_length_limit,
+            $banner_container,
+            $("#message-content-container"),
+        );
```

Comment:
Does this generate a banner, or a tooltip? The issue specification mentions a tooltip, and this looks like a banner to me.